File: pubnub-java/src/main/java/com/pubnub/api/v2/callbacks/StatusListener.java
package com.pubnub.api.v2.callbacks;

import com.pubnub.api.PubNub;
import com.pubnub.api.models.consumer.PNStatus;
import com.pubnub.internal.BasePubNub;
import org.jetbrains.annotations.NotNull;

public abstract class StatusListener implements com.pubnub.internal.v2.callbacks.StatusListener {
    public abstract void status(@NotNull PubNub pubnub, @NotNull PNStatus pnStatus);

    @Override
    public void status(@NotNull BasePubNub pubnub, @NotNull PNStatus status) {
        status((PubNub) pubnub, status);
    }
}

File: pubnub-kotlin-core/src/main/kotlin/com/pubnub/internal/v2/callbacks/StatusListener.kt
package com.pubnub.internal.v2.callbacks

import com.pubnub.api.models.consumer.PNStatus
import com.pubnub.internal.BasePubNub
import com.pubnub.internal.callbacks.Listener

interface StatusListener : Listener {
    /**
     * Receive status updates from the PubNub client.
     *
     * @see [PNStatus]
     *
     * @param pubnub The client instance which has this listener attached.
     * @param status Wrapper around the actual message content.
     */
    fun status(pubnub: BasePubNub, status: PNStatus)
}
